include/dao/db_migrate.js

```diff
@@ -0,0 +1,138 @@
+        'section_map',
+        'email_settings'
+    ];
+
+    function DBMigrate() {
+
+        this.run = function (cb) {
+            var self = this;
+            var siteService = new pb.SiteService();
+            siteService.getSiteMap(function (err, result) {
+                if (pb.config.multisite && result.active.length === 0 && result.inactive.length === 0) {
+                    self.createSite(function (err, isTaken, field, result) {
+                        self.siteUid = result.uid;
+                        var tasks = [
+                            util.wrapTask(self, self.migrateContentAndPluginData),
+                            util.wrapTask(self, self.migrateSettings),
+                            util.wrapTask(self, self.migrateUsers)
+                        ];
+                        //self.migrateContentAndPluginData(cb);
+                        async.series(tasks, function(err, result) {
+                           cb(err, result);
+                        });
+                    });
+                }
+                else {
+                    cb(null, true);
+                }
+            });
+        };
+
+        this.createSite = function (cb) {
+            var siteService = new pb.SiteService();
+            var site = pb.DocumentCreator.create('site', {
+                displayName: pb.config.siteName,
+                hostname: url.parse(pb.config.siteRoot).host
+            });
+            siteService.createSite(site, '', cb);
+        };
+
+        this.migrateContentAndPluginData = function(cb) {
+            var self = this;
+            var tasks = pb.util.getTasks(MIGRATE_ALL, function (collections, i) {
+                return function (callback) {
+                    self.migrateCollection(collections[i], self.siteUid, callback);
+                };
+            });
+
+            async.parallel(tasks, function (err, result) {
+                cb(err, result);
+            });
+        };
+
+        this.migrateSettings = function (cb) {
+            this.migrateGlobalSubCollection('setting', SITE_SPECIFIC_SETTINGS, 'key', cb);
+        };
+
+        this.migrateUsers = function(cb) {
+            this.migrateGlobalSubCollection('user', SITE_SPECIFIC_USERS, 'admin', cb);
+        };
+
+        this.migrateGlobalSubCollection = function(collection, siteSpecificArr, compareTo, cb) {
+            var self = this;
+            var dao = new pb.DAO();
+            dao.q(collection, {}, function(err, results) {
+                var tasks = util.getTasks(results, function(results, i) {
+                    return function(callback) {
+                      var uid = siteSpecificArr.indexOf(results[i][compareTo]) > -1? self.siteUid : pb.SiteService.GLOBAL_SITE;
+                      self.applySiteToDocument(results[i], uid, callback);
+                    };
+                });
+
+                async.parallel(tasks, function(err, result) {
+                    cb(err, result);
+                });
```

Comment:
Done.

Ben Tidwell
Software Engineer III | Talent Network
CareerBuilder
5550-A Peachtree Pkwy, Norcross GA 30092

From: Brian <notifications@github.com<mailto:notifications@github.com>>
Reply-To: pencilblue/pencilblue <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Sunday, July 12, 2015 at 7:24 PM
To: pencilblue/pencilblue <pencilblue@noreply.github.com<mailto:pencilblue@noreply.github.com>>
Cc: Ben Tidwell <ben.tidwell@careerbuilder.com<mailto:ben.tidwell@careerbuilder.com>>
Subject: Re: [pencilblue] Multi Tenancy (#680)


In include/dao/db_migrate.js<https://github.com/pencilblue/pencilblue/pull/680#discussion_r34426968>:

> +        };
> +
> +        this.migrateGlobalSubCollection = function(collection, siteSpecificArr, compareTo, cb) {
> +            var self = this;
> +            var dao = new pb.DAO();
> +            dao.q(collection, {}, function(err, results) {
> +                var tasks = util.getTasks(results, function(results, i) {
> +                    return function(callback) {
> +                      var uid = siteSpecificArr.indexOf(results[i][compareTo]) > -1? self.siteUid : pb.SiteService.GLOBAL_SITE;
> +                      self.applySiteToDocument(results[i], uid, callback);
> +                    };
> +                });
> +
> +                async.parallel(tasks, function(err, result) {
> +                    cb(err, result);
> +                });


It is redundant to provide an anonymous function then use the callback. Just pass the callback to parallel

async.parallel(tasks, cb);


-
Reply to this email directly or view it on GitHub<https://github.com/pencilblue/pencilblue/pull/680/files#r34426968>.